packages/taco/src/taco-client.ts

```diff
@@ -0,0 +1,322 @@
+   *
+   * @param {string | Uint8Array} data - String or Uint8Array to encrypt
+   * @param {Condition} accessCondition - Access condition for decryption
+   * @param {DkgPublicKey} dkgPublicKey - The DKG public key to use for encryption
+   * @returns {Promise<ThresholdMessageKit>} Encrypted message kit
+   * @throws {Error} If encryption fails
+   *
+   * @example
+   * ```typescript
+   * // Get DKG public key from ritual or cache
+   * const dkgPublicKey = await getDkgPublicKey(domain, ritualId);
+   *
+   * // Encrypt offline using the public key
+   * const messageKit = await tacoClient.encryptWithPublicKey('Hello, secret!', condition, dkgPublicKey);
+   * ```
+   */
+  async encryptWithPublicKey(
+    data: string | Uint8Array,
+    accessCondition: Condition,
+    dkgPublicKey: DkgPublicKey,
+  ): Promise<ThresholdMessageKit> {
+    await TacoClient.initialize();
+
+    try {
+      const messageKit = await encryptWithPublicKey(
+        data,
+        accessCondition,
+        dkgPublicKey,
+        (this.config as TacoClientEthersConfig).ethersSigner ||
+          (this.config as TacoClientViemConfig).viemSignerAccount,
+      );
+
+      return messageKit;
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : String(error);
+      throw new Error(
+        `TACo encryption with public key failed: ${errorMessage}`,
+      );
+    }
+  }
+
+  /**
+   * Decrypt data using TACo
+   *
+   * @param {ThresholdMessageKit | Uint8Array} encryptedData - Either a ThresholdMessageKit or encrypted bytes (Uint8Array)
+   * @param {ConditionContext} [conditionContext] - Optional condition context for time-based conditions
+   * @returns {Promise<Uint8Array>} Decrypted data
+   * @throws {Error} If decryption fails
+   *
+   * @example
+   * ```typescript
+   * // With messageKit
+   * const decrypted = await tacoClient.decrypt(messageKit, conditionContext);
+   *
+   * // With encrypted bytes
+   * const decrypted = await tacoClient.decrypt(encryptedBytes, conditionContext);
+   * ```
+   */
+  async decrypt(
+    encryptedData: ThresholdMessageKit | Uint8Array,
+    conditionContext?: ConditionContext,
+  ): Promise<Uint8Array> {
+    await TacoClient.initialize();
+
+    // Handle both messageKit and encrypted bytes
+    const messageKit =
+      encryptedData instanceof ThresholdMessageKit
+        ? encryptedData
+        : ThresholdMessageKit.fromBytes(encryptedData);
+
+    try {
+      const decrypted = await decrypt(
+        (this.config as TacoClientEthersConfig).ethersProvider ||
+          (this.config as TacoClientViemConfig).viemClient,
+        this.config.domain,
+        messageKit,
+        conditionContext,
+        this.config.porterUris,
+      );
+
+      return decrypted;
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : String(error);
+      throw new Error(`TACo decryption failed: ${errorMessage}`);
+    }
+  }
+
```

Comment:
I see in https://github.com/nucypher/sprints/issues/217 that you mention a function called `decryptWithAutoContext` but I didn't find such function in the code. Just curious about it. Have we ruled it out?